subscribe.go

```diff
@@ -34,19 +37,33 @@ type legSubscriber struct {
 
 // NewSubscriber creates a new leg subscriber listening to a specific pubsub topic
 // with a specific policyHandle to determine when to perform exchanges
-func NewSubscriber(ctx context.Context, dt *LegTransport,
-	policy PolicyHandler) (LegSubscriber, error) {
-	return newSubscriber(ctx, dt, policy)
+func NewSubscriber(ctx context.Context,
```

Comment:
should we support a `NewSubscriberFromExisting`?

Reply:
I don't think until we have a use case